src/analysis/specific_limits.lean

```diff
@@ -15,6 +15,72 @@ open classical function lattice filter finset metric
 
 variables {α : Type*} {β : Type*} {ι : Type*}
 
+/-- If a function tends to infinity along a filter, then this function multiplied by a positive
+constant (on the left) also tends to infinity. The archimedean assumption is convenient to get a
+statement that works on `ℕ`, `ℤ` and `ℝ`, although not necessary (a version in ordered fields is
+given in `tendsto_at_top_mul_left'`). -/
+lemma tendsto_at_top_mul_left [decidable_linear_ordered_semiring α] [archimedean α]
+  {l : filter β} {r : α} (hr : 0 < r) {f : β → α} (hf : tendsto f l at_top) :
+  tendsto (λx, r * f x) l at_top :=
+begin
+  apply (tendsto_at_top _ _).2 (λb, _),
+  obtain ⟨n, hn⟩ : ∃ (n : ℕ), (1 : α) ≤ n • r := archimedean.arch 1 hr,
+  have hn' : 1 ≤ r * n, by rwa add_monoid.smul_eq_mul' at hn,
+  filter_upwards [(tendsto_at_top _ _).1 hf (n * max b 0)],
+  assume x hx,
+  calc b ≤ 1 * max b 0 : by { rw [one_mul], exact le_max_left _ _ }
+  ... ≤ (r * n) * max b 0 : mul_le_mul_of_nonneg_right hn' (le_max_right _ _)
+  ... = r * (n * max b 0) : by rw [mul_assoc]
+  ... ≤ r * f x : mul_le_mul_of_nonneg_left hx (le_of_lt hr)
+end
+
+/-- If a function tends to infinity along a filter, then this function multiplied by a positive
+constant (on the right) also tends to infinity. The archimedean assumption is convenient to get a
+statement that works on `ℕ`, `ℤ` and `ℝ`, although not necessary (a version in ordered fields is
+given in `tendsto_at_top_mul_right'`). -/
+lemma tendsto_at_top_mul_right [decidable_linear_ordered_semiring α] [archimedean α]
```

Comment:
Honest question: are there archimedean linearly ordered semirings that are not commutative?

Reply:
Honest answer: I don't know, but I think all the interesting examples are commutative. The reason for this formulation is that I could not find the class `decidable_linear_ordered_comm_semiring`!

Reply:
Polynomials in two non-commuting variables with some lexicographic norm trick?